hadoop-crypto/src/main/java/com/palantir/crypto2/hadoop/EncryptedFileSystem.java

```diff
@@ -172,9 +172,14 @@ public boolean delete(Path path, boolean recursive) throws IOException {
             throw new UnsupportedOperationException("EncryptedFileSystem does not support recursive deletes");
         }
 
-        // Interrupted deletes should be resumable. They are expected to be retried.
-        tryRemoveKey(path);
-        return fs.delete(path, false);
+        boolean success = fs.delete(path, false);
+
+        if (success || !fs.exists(path)) {
+            // Interrupted deletes should be resumable. They are expected to be retried.
+            tryRemoveKey(path);
```

Comment:
I think in both `create` and `encrypt` we 1) create the file and then 2) store the key material, so I believe this is doing the inverse operation already by removing the key first and then deleting the file?

Reply:
Maybe my question is: what does inverting these calls fix exactly? as soon as either of the file or key material are deleted the file should no longer be readable, so I image there's some other weird error state we're trying to kill?

Reply:
Hmm fair, what happens is sort of in the middle here. `create` and `encrypt` create an empty file before they add the key material, but they don't actually return the file and thus add contents to it yet. The key material file would get created before either method returns, and thus before any contents get added to the file.

We've recently seen issues where a file existed with its full content, but without its associated key material file, so I'm trying to remove all ways for that to happen.